examples/zh/helloWorld.md

```diff
@@ -0,0 +1,303 @@
+[base]: ../../docs/glossary.md#base
+[config]: https://github.com/kinflate/example-hello
+[gitops]: ../../docs/glossary.md#gitops
+[hello]: https://github.com/monopole/hello
+[kustomization]: ../../docs/glossary.md#kustomization
+[original]: https://github.com/kinflate/example-hello
+[overlay]: ../../docs/glossary.md#overlay
+[overlays]: ../../docs/glossary.md#overlay
+[patch]: ../../docs/glossary.md#patch
+[variant]: ../../docs/glossary.md#variant
+[variants]: ../../docs/glossary.md#variant
+
+# Demo: hello world with variants
+
+步骤：
+
+ 1. 拉取已经存在的的 [base] 配置。
+ 1. 进行定制。
+ 1. 基于定制后的 base 新建2个不同的 [overlays] (_staging_ 和 _production_)。
+ 1. 运行 kustomize 和 kubectl 来部署 staging 和 production 。
+
+首先创建一个工作空间：
+
+<!-- @makeWorkplace @test -->
+```
+DEMO_HOME=$(mktemp -d)
+```
+
+或者：
+
+> ```
+> DEMO_HOME=~/hello
+> ```
+
+## 创建 base
+
+让我们开始运行 [hello] 服务。
+
+如果要使用 [overlays] 创建 [variants] ，必须先创建一个共同的 [base] 。
+
+为了使本文档保持简洁，base 的资源位于补充目录中，并不在此处，请按照下面的方法下载它们：
+
+<!-- @downloadBase @test -->
+```
+BASE=$DEMO_HOME/base
+mkdir -p $BASE
+
+curl -s -o "$BASE/#1.yaml" "https://raw.githubusercontent.com\
+/kubernetes-sigs/kustomize\
+/master/examples/helloWorld\
+/{configMap,deployment,kustomization,service}.yaml"
+```
+
+观察该目录：
+
+<!-- @runTree @test -->
+```
+tree $DEMO_HOME
+```
+
+可以看到：
+
+> ```
+> /tmp/tmp.IyYQQlHaJP
+> └── base
+>     ├── configMap.yaml
+>     ├── deployment.yaml
+>     ├── kustomization.yaml
+>     └── service.yaml
+> ```
+
+这些 resources 可以立即在 k8s 集群中部署：
+
+> ```
+> kubectl apply -f $DEMO_HOME/base
+> ```
+
+实例化 _hello_ 服务， `kubectl` 只能识别 resources 文件。
+
+
+### The Base Kustomization
+
+`base` 目录中包含一个 [kustomization] 文件：
+
+<!-- @showKustomization @test -->
+```
+more $BASE/kustomization.yaml
+```
+
+（可选）在 base 上运行 `kustomize` 将自定义 resources 输出到 `stdout` ：
```

Comment:
This sentence is confusing
```
在 base 上运行 `kustomize` 将自定义 resources 输出到 `stdout`
```
How about
```
在 base 目录上运行 `kustomize` 将定制过的 resources 打印到标准输出
```
I couldn't find a better translation for `customized`.